blocks/extensions/saveload.py

```diff
@@ -94,12 +74,20 @@ def do(self, callback_name, *args):
             path = self.path
             if from_user:
                 path, = from_user
-            secure_dump(self.main_loop, path, use_cpickle=self.use_cpickle)
-            filenames = self.save_separately_filenames(path)
-            for attribute in self.save_separately:
-                secure_dump(getattr(self.main_loop, attribute),
-                            filenames[attribute], cPickle.dump,
-                            protocol=DEFAULT_PROTOCOL)
+            if self.save_separately:
+                to_add = {}
+                for attr in self.save_separately:
+                    to_add[attr] = getattr(self.main_loop, attr)
+            else:
+                to_add = None
+            if self.parameters is None:
+                if hasattr(self.main_loop, 'model'):
+                    self.parameters = self.main_loop.model.parameters
+            secure_dump(self.main_loop, path,
+                        dump_function=_dump_and_add_to_dump,
+                        parameters=self.parameters,
+                        to_add=to_add,
+                        use_cpickle=self.use_cpickle)
```

Comment:
Seems like `pickle_object` can not be used here, which is not nice. Many people are willing to sacrifice pickling the whole main loop in favour of having less restrictions on their code.